x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java

```diff
@@ -104,6 +104,10 @@ public class EsqlFeatures implements FeatureSpecification {
      * Cast string literals to a desired data type.
      */
     public static final NodeFeature STRING_LITERAL_AUTO_CASTING = new NodeFeature("esql.string_literal_auto_casting");
+    /**
+     * Support multiple field mappings if appropriate conversion function is used (union types)
+     */
+    public static final NodeFeature UNION_TYPES = new NodeFeature("esql.union_types");
```

Comment:
This sounds like a solution to a problem we suspect we might get in future, but do not (yet) suffer from, the issue of the creation of too many NodeFeatures. I think there are many solutions to that problem, like regular removal of older NodeFeature instances and references to them, since they should only really matter for BWC and rollover of recent releases. Or some other solution... let's not make it part of this PR (or any feature PR).

Reply:
I'm with @craigtaverner on this. I'd prefer not to adopt a new pattern here. I'm not really sure what the pattern should be either. Or if we need one.